Studs/Events/ViewModels/EventsViewModel.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  EventsViewModel.swift
+//  Studs
+//
+//  Created by Willy Liu on 2019-02-22.
+//  Copyright © 2019 Studieresan. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+
+final class EventsViewModel {
+
+    // MARK: Properties
+
+    let sectionTitles = ["Next event", "Upcoming", "Past events"]
+
+    // Array of next event, upcoming events, past events
+    private var events = [[Event](), [Event](), [Event]()]
+
+    // Array of next event cellVM, upcoming event cellVMs, past event cellVMs
+    private var cellViewModels: [[EventCellViewModel]] =
+        [[EventCellViewModel](), [EventCellViewModel](), [EventCellViewModel]()] {
+        didSet {
+            self.reloadTableViewClosure?()
+        }
+    }
+
+    private let disposeBag = DisposeBag()
+
+    // What should happen when the cellVMs are changed
+    var reloadTableViewClosure: (() -> Void)?
+
+    // MARK: Public methods
+
+    func fetchData() {
+        // Fetch the events
+        Http.fetchAllEvents().subscribe(onNext: { events in
```

Comment:
You probably want a weak reference to `self` here. Imagine what happens when the network request starts and the user leaves the view.